proposed addition of a df.agg stacked go.bar example (#4546)
plotly/plotly.py: doc/python/bar-charts.md

fig.show()
```

> ### Stacked Bar Chart from aggregating a data frame 

Reviewer: ```suggestion
### Stacked Bar Chart From Aggregating a DataFrame 
```

> Stacked bar charts are a powerful way to present results summarizing categories generated using the Pandas aggregate commands.  DF.agg() which produces a wide format data set with one row for each bar component and a column for each bar, which is the transpose of the orientation of the px.bar wide data frame.  Tranposing and updating the indexes is a somewhat involved option. Here is one straightforward way to aggregate a data set into a summarized form and present the results as a stacked bar.
> 

Reviewer: ```suggestion
Stacked bar charts are a powerful way to present results summarizing categories generated using the Pandas aggregate commands. `pandas.DataFrame.agg` produces a wide format data set with one row for each bar component and a column for each bar, which is the transpose of the orientation of the `px.bar` wide data frame.  Transposing and updating the indexes is a somewhat involved option. Here is one straightforward way to aggregate a data set into a summarized form and present the results as a stacked bar using `plotly.graph_objects`.

```

Thanks for the PR @rl-utility-man! Great example. 
I edited to provide some full names of the functions. 

Do you think changing the following 

```
`pandas.DataFrame.agg` produces a wide format data set with one row for each bar component and a column for each bar, which is the transpose of the orientation of the `px.bar` wide data frame.  
```
to
```
`pandas.DataFrame.agg` produces a wide format data set that doesn't work directly with `px.bar`.
```
retains the meaning you want to convey?

Author: Glad you liked this @LiamConnors!  Yes, shortening that detailed sentence is an improvement. Consider getting the reversed/transposed idea in succinctly.

How about:

`pandas.DataFrame.agg` produces a wide format data set with the rows and columns reversed from the `px.bar` expectation.

Reviewer: Thanks @rl-utility-man. That sounds good to me. I've updated my commit suggestion based on it. What do you think?

Author: Looks great @LiamConnors.  My only remaining suggestion is to rewrite the last sentence to:

 Here is one straightforward alternative, which presents the aggregated data as a stacked bar using `plotly.graph_objects`.

```

from plotly import graph_objects as go
import pandas as pd

> #get one year of gapminder data
> url = 'https://raw.githubusercontent.com/plotly/datasets/master/gapminderDataFiveYear.csv'
> df = pd.read_csv(url)
> df = df[df['year']==2007]
> df["gdp"]=df["pop"]*df['gdpPercap']
> 
> 
> #build the summary of interest
> df_summarized = df.groupby("continent", observed=True).agg("sum").reset_index()
> 
> df_summarized["percent of world population"]=100*df_summarized["pop"]/df_summarized["pop"].sum()
> df_summarized["percent of world GDP"]=100*df_summarized["gdp"]/df_summarized["gdp"].sum()
> 
> 
> df = df_summarized[["continent", 
> "percent of world population",
> "percent of world GDP",
> ]]
> 
> #we now have a wide data frame, but it's in the opposite orientation from the one that px is designed to deal with.
> #transposing it and rebuilding the indexes is an option, but iterating through the DF using graph objects is more succinct. 
> 
> fig=go.Figure()
> for category in df_summarized["continent"].values:
>     fig.add_trace(go.Bar(
>             x=df.columns[1:],
>             #we need to get a pandas series that contains just the values to graph; 
>             #we do so by selecting the right row, selecting the right columns
>             #and then tranposing and using iloc to convert to a series
>             #here, I assume that the bar element category variable is in column 0
>             y=list(df.loc[df["continent"]==category][list(df.columns[1:])].transpose().iloc[:,0]),
>             name=str(category)
> 
> 

Reviewer: ```suggestion
# Get one year of gapminder data
url = 'https://raw.githubusercontent.com/plotly/datasets/master/gapminderDataFiveYear.csv'
df = pd.read_csv(url)
df = df[df['year']==2007]
df["gdp"]=df["pop"]*df['gdpPercap']

# Build the summary of interest
df_summarized = df.groupby("continent", observed=True).agg("sum").reset_index()

df_summarized["percent of world population"]=100*df_summarized["pop"]/df_summarized["pop"].sum()
df_summarized["percent of world GDP"]=100*df_summarized["gdp"]/df_summarized["gdp"].sum()

df = df_summarized[["continent", 
"percent of world population",
"percent of world GDP",
]]

# We now have a wide data frame, but it's in the opposite orientation from the one that px is designed to deal with.
# Transposing it and rebuilding the indexes is an option, but iterating through the DF using graph objects is more succinct. 

fig=go.Figure()
for category in df_summarized["continent"].values:
    fig.add_trace(go.Bar(
            x=df.columns[1:],
            # We need to get a pandas series that contains just the values to graph; 
            # We do so by selecting the right row, selecting the right columns
            # and then transposing and using iloc to convert to a series
            # Here, we assume that the bar element category variable is in column 0
            y=list(df.loc[df["continent"]==category][list(df.columns[1:])].transpose().iloc[:,0]),
            name=str(category)

```

Reviewer: This looks like a big change, but it's mostly just suggesting changing spacing at the start of the comments for readability.

Author: Looks great!  Much appreciated!
